staging/src/k8s.io/sample-controller/main.go

```diff
@@ -58,8 +58,23 @@ func main() {
 		klog.Fatalf("Error building example clientset: %s", err.Error())
 	}
 
-	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, time.Second*30)
-	exampleInformerFactory := informers.NewSharedInformerFactory(exampleClient, time.Second*30)
+	// Informers trigger events on resource changes, which typically queues
+	// a reconciliation of that resource.  The resync period acts as a
+	// safety net to protect against dropped events, and the informer will
```

Comment:
"Acts as a safety net" sounds like a positive thing. "Can be used to limp along in the presence of controller bugs" sounds more like how I feel, anyway. :)

This comment doesn't capture some details I think are important:

1. A legitimate usage for this is polling non-k8s resources (e.g., Load Balancers in your cloud provider). Choose the period based on responsiveness vs cloud-provider (NOT k8s) qps usage.
2. Ordinary network failures etc means that your controller COULD get a resync at any time. You must test and handle this. A small value here in tests ensures that this critical code path is exercised.
3. Consider e.g. setting a metric or logging whenever your controller takes action on a resync. Unless you're polling some other resource, it's probably a bug.